qemu/tests/passt_transfer_tests.py

```diff
@@ -0,0 +1,121 @@
+        gateway = utils_net.get_default_gateway(ip_ver='ipv4')
+        local_address = "localhost"
+    # prepare test data
+    guest_path = (tmp_dir + "src-%s" % utils_misc.generate_random_string(8))
+    dest_path = (tmp_dir + "dst-%s" % utils_misc.generate_random_string(8))
+    host_path = os.path.join(test.tmpdir, "tmp-%s" %
+                             utils_misc.generate_random_string(8))
+    test.log.info("Test setup: Creating %dbytes file on host", filesize)
+    process.run(dd_cmd % (host_path, filesize), shell=True)
+
+    try:
+        src_md5 = (crypto.hash_file(host_path, algorithm="md5"))
+        error_context.context("md5 value of data from src: %s" % src_md5,
+                              test.log.info)
+        # transfer data from host to guest
+        host2guest_port = 10001
+        for vm in vms:
+            error_context.context("Transfer data from host to %s" % vm.name,
+                                  test.log.info)
+            sessions[vm].cmd_output_safe(fw_stop_cmd)
+            sessions[vm].sendline(receive_cmd % (host2guest_port, guest_path))
+            time.sleep(5)
+            process.run(sent_cmd % (host_path, local_address, host2guest_port), shell=True)
+            dst_md5 = get_file_md5sum(guest_path, sessions[vm],
+                                      timeout=file_md5_check_timeout)
+            error_context.context("md5 value of data in %s: %s" % (vm.name, dst_md5),
+                                  test.log.info)
+            if dst_md5 != src_md5:
+                test.fail("File changed after transfer host -> %s" % vm.name)
+            host2guest_port += 1
+
+        # transfer data between guest2guest
+        guest2guest_port = 10001
+        for vm1 in vms:
+            for vm2 in vms:
+                if vm1 != vm2:
+                    error_context.context("Transferring data from %s to %s" %
+                                          (vm1.name, vm2.name), test.log.info)
+                    sessions[vm1].sendline(receive_cmd % (guest2guest_port, dest_path))
+                    time.sleep(5)
+                    sessions[vm2].sendline(sent_cmd % (guest_path, gateway, guest2guest_port))
+                    dst_md5 = get_file_md5sum(dest_path, sessions[vm1],
+                                              timeout=file_md5_check_timeout)
+                    error_context.context("md5 value of data in %s: %s" % (vm.name, dst_md5),
+                                          test.log.info)
+                    if dst_md5 != src_md5:
+                        test.fail("File changed transfer %s -> %s" % (vm1.name, vm2.name))
+            guest2guest_port += 1
+
+    finally:
+        process.system("rm -rf %s" % host_path, timeout=timeout,
+                       ignore_status=True)
+        for vm in vms:
+            sessions[vm].cmd("rm -rf %s %s || true" % (guest_path, dest_path),
+                             timeout=timeout, ignore_all_errors=True)
+            sessions[vm].close()
```

Comment:
You use `image_snapshot = yes`, so you can skip this.

Reply:
Thanks for suggestion, it will be change at next update.